ui/search.js

```diff
@@ -53,17 +58,48 @@ function removeCloseButton() {
     delete searchResults._closeButton;
 }
 
-function setSearchResultsXAlign(align) {
+function addMaxWidthBoxInSearch() {
     const { viewSelector } = Main.overview;
-    viewSelector._searchResults.x_align = align;
+
+    viewSelector._searchPage.remove_child(viewSelector._searchResults);
+
+    const maxWidthBox = new Search.MaxWidthBox({ name: 'searchResultsBox' });
+    maxWidthBox.add_child(viewSelector._searchResults);
+
+    viewSelector._searchPage.child = maxWidthBox;
+}
+
+function removeMaxWidthBoxFromSearch() {
+    const { viewSelector } = Main.overview;
+
+    const maxWidthBox = viewSelector._searchResults.get_parent();
+    maxWidthBox.remove_child(viewSelector._searchResults);
+
+    viewSelector._searchPage.child = viewSelector._searchResults;
+    maxWidthBox.destroy();
 }
 
 function enable() {
-    setSearchResultsXAlign(Clutter.ActorAlign.CENTER);
+    Utils.override(Search.ProviderInfo, '_init', function(provider) {
```

Comment:
I'm a bit ware of overriding `_init` as this only works for new objects created - but as long as we can guarantee these objects will be created on demand I think it should be fine.

Reply:
You can check the `search.js` file and verify that these are created on demand.

Reply:
I did and it was not clear if that is the case, maybe we should invoke `_reloadRemoteProviders()` which will ensure a new view is created and in turn the overridden ProviderInfo.

Reply:
If we call _reloadRemoteProviders on enable we should do the same on disable.

Reply:
How does someone enable or disable this extension while the search view is open, and has results being displayed?

Reply:
As I said it was not clear to me whether the search view would always be recreated when starting a search, but if you know that it is the case then we are good.